### DISCUSSION TITLE
Unable to connect to metamask using iphone mobile chrome browser

### DISCUSSION THREAD

**Discussion by guiguy2018** [WalletConnect/walletconnect-monorepo#3232] [Developer Support] [answered]
Using
web3modal 2.6.2
Mobile Metamask IOS v7.2.0(1142)
Mobile Chrome IOS 114.0.5735.124
iPhone IOS 16.5.1

From Mobile phone chrome, https://web3modal.com/, hit connect. Then connect your wallet popup, select Metamak. Metamask launches but the “pop up to connect to an address” does not appear 99% of the time.

Scan qr code from another device metamask wallet does work.

I tested with Uniswap and Trust wallets and they work much better.

**Comment by BritvaBo** [comment]
Had some problem. Thanks

**Comment by glitch-txs** [comment]
Hi is this still happening on https://web3modal.com/ for you? I'm not to able to reproduce.

Do you have other extension wallets installed?

**Reply by guiguy2018** [reply]
happens on ios mobile chrome, works fine on desktops with wallet extensions.

**Reply by glitch-txs** [reply]
Thanks for the update, could you test this on labs please?
https://lab.web3modal.com/

**Reply by guiguy2018** [reply]
Still same problem.  I cannot connect with metamask on ios mobile.  I can cannect with the Uniswap wallet.  I tried all the buttons on lab.web3modal.com

**Reply by glitch-txs** [reply]
Which chain are you in the wallet when connecting?
Also could you update to the latest version

**Reply by guiguy2018** [reply]
Using Ethereum mainnet
IOS 14.6
IOS Metamask v7.3.0(1144)
I believe that's the latest updates and I've cleared the cache.  Same problem with Chrome and Safari browser.

**Reply by glitch-txs** [reply]
Do you see any errors in the website console?

**Reply by guiguy2018** [reply]
In IOS chrome mobile, I can only access the chrome logs using chrome://inspect which gives me this
INFO [object object]
INFO [object object]

**Comment by ghost** [comment]
To connect to the site from your phone, you need to access the site from the metamask application itself.

**Reply by guiguy2018** [reply]
yes, it works from the browser within Metamask.  I would like it to work with my dapp from the mobile browser.  It works with the Uniswap wallet so I think it's a Metamask thing.

**Reply by glitch-txs** [reply]
with WalletConnect you should be able to connect inside the app and from the outside as well

**Reply by guiguy2018** [reply, marked as answer]
Got it working on mobile devices.  It seems that an ad ad-blocker was blocking api2.block.io.  I don't fully understand what api2.branch.io does, but it all started working once api2.branch.io was whitelisted.

I still don't understand why it worked using the embedded chrome browser in Metamask and not the mobile chrome browser until it unblocked api2.branch.io

**Reply by temakz** [reply]
rainbow wallet

**Comment by turtlewuffle** [comment]
a couple of times on android there was a similar problem

**Comment by sergeyskripko35** [comment]
Had a similar problem... Need to change wallet
